### PR TITLE
Feat: add POAP logic

### PR DESCRIPTION
**Pull request by ghost**
Added POAP logic

Changes made:
Added check if user holds the POAP token that we need (we are currently using a test token. I will send it along with a mint link for testing)
Removed token gating code as it will no longer be used

Expected behaviour:
"Claim Here" button on shopify window should be disabled if the user does not hold the POAP token or their wallet is not connected.

If the user holds the correct POAP token, the "Claim Here" button should be enabled. Nothing currently happens when you click the button as I am working on that logic in a separate PR.

### CODE REVIEW

File: components/shopify-widget/index.tsx [hunk 1]
```diff
@@ -57,11 +85,19 @@ const ShopifyWidget = () => {
                 </Text>
               </Flex>
               <PsyButton
-                onClick={() => {
-                  console.log("insert claim logic");
+                onClick={async () => {
```
Comment (gaelicgod) [inline components/shopify-widget/index.tsx:88]: if this ready for review, shouldn't this be removed?

File: pages/api/poap.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      );
+      if (poapRes.status === 404) {
+        console.info("User does not hold this POAP.");
+        res.status(201).send({
+          message: "User does not hold a valid POAP token"
+        });
+      }
+
+      if (poapRes.status !== 404 && !poapRes.ok) {
+        res.status(poapRes.status).send(poapRes.statusText);
```
Comment (gaelicgod) [inline pages/api/poap.ts:82]: add a `return` like above

File: pages/api/shopify.ts [hunk 3]
```diff
@@ -29,16 +28,6 @@ const SHOPIFY_API_SECRET = env.SHOPIFY_API_SECRET;
...
-  ? psyNFTMainnet
-  : psyNFTSepolia;
-const ETHEREUM_RPC_URL = env.NEXT_PUBLIC_MAINNET_CLIENT;
-
-const publicClient = createPublicClient({
-  chain: mainnet,
-  transport: http(ETHEREUM_RPC_URL)
-});
-
 console.debug("SHOPIFY_API_ACCESS_TOKEN => ", SHOPIFY_API_ACCESS_TOKEN);
```
Comment (gaelicgod) [inline pages/api/shopify.ts:31]: can we remove this now

File: pages/api/poap.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          headers: {
+            accept: "application/json",
+            "X-API-KEY": env.POAP_API_KEY
+          },
+          method: "GET"
+        }
+      );
+      if (poapRes.status === 404) {
+        console.info("User does not hold this POAP.");
+        res.status(201).send({
```
Comment (gaelicgod) [inline pages/api/poap.ts:76]: why do you send a `201` response? `201` is the success message for a resource creation

**Review by gaelicgod — CHANGES_REQUESTED** [review]
Please see comments

File: pages/api/poap.ts (hunk 4 above)
Comment (EsterB12) [reply pages/api/poap.ts]: I changed this to a 204 response - which is No Content
is this alright?

File: components/shopify-widget/index.tsx [hunk 5]
```diff
@@ -57,11 +85,18 @@ const ShopifyWidget = () => {
...
+                customStyle={{
+                  width: "100%"
                 }}
+                isDisabled={!userHoldsPOAPToken}
               >
-                Claim Here
+                {address
+                  ? userHoldsPOAPToken
+                    ? "Claim Here"
+                    : "Cannot Claim"
```
Comment (gaelicgod) [inline components/shopify-widget/index.tsx:98]: is this the copy from the design?

Comment (EsterB12) [reply components/shopify-widget/index.tsx:98]: This change was made based on this interaction
https://linumlabs-workspace.slack.com/archives/C06MT4775FE/p1728033867703989

but I can change the copy if needed?

**Review by gaelicgod — APPROVED** [review]

**Review by gaelicgod — APPROVED** [review]

**Pull request merged**